crates/gitbutler-core/src/virtual_branches/virtual.rs

```diff
@@ -2122,12 +2120,21 @@ pub fn write_tree_onto_tree(
                         filemode = git::FileMode::BlobExecutable;
                     }
                 }
+
                 #[cfg(target_os = "windows")]
                 {
-                    // TODO(qix-): Pull from `core.filemode` config option to determine
-                    // TODO(qix-): the behavior on windows. For now, we set this to true.
-                    // TODO(qix-): It's not ideal, but it gets us to a windows build faster.
-                    filemode = git::FileMode::BlobExecutable;
+                    // NOTE: *Keep* the existing executable bit if it was present
+                    //       in the tree already, don't try to derive something from
+                    //       the FS that doesn't exist.
```

Comment:
This looks good, but is there a way to have `filemode` (line 2120) pull from the config, too?

Reply:
I don't think it's applicable as we do not get an 'executable' bit with any Rust API. If we would get that bit, `core.filemode = false` would mean we ignore it, but there is no way to get it in the first place.

Or in other words, `core.filemode` doesn't mean that everything is executable by default, but I don't have a Windows installation ready yet to validate this.

Reply:
Hmm alright. Well we'll see if anyone has an issue with this behavior then.

Reply:
Yes, it's a bit strange. Interestingly I never had that problem with `gitoxide` as it can't alter the index yet and add files to it, which would have run into this as well.